### ISSUE TITLE
 Support PlusCal await statement 

### ISSUE THREAD

**Issue by mrordinaire** [UBC-NSS/pgo#10]
In PlusCal, await blocks an atomic step from executing until the condition is true. The following might be a way to implement await in Go:

- Enforce that await can only occur at the top of an atomic step. (Otherwise we have to deal with making temporary variables since part of the step may mutate global state.)
- Make sync.Conds for each variable/each condition.
- When a variable is modified, wake up the corresponding sync.Conds. There could be a global handler which waits on all sync.Conds and wakes up goroutines accordingly, or the goroutines could deal with checking the condition themselves.

Original issue: https://bitbucket.org/whiteoutblackout/pgo/issues/38/support-pluscal-await-statement

**Comment by mrordinaire** [comment]
After some discussion, we have agreed to put in the first restriction, i.e. await statements can only occur at the top of an atomic step.

Await statement
``` PlusCal
await exp;
```
will be translated into
``` Go
for {
    lock
    if exp {
        break
    }
    unlock
    yield
}
// use variables
unlock
```
The yield statement in the above infinite loop is to invoke the Go scheduler.

Original comment: https://bitbucket.org/whiteoutblackout/pgo/issues/38/support-pluscal-await-statement#comment-42541651

**Comment by mrordinaire** [comment]
Fixed in 6a8a4a7ba5
